packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -137,7 +137,7 @@ export function isSyntheticShadowRoot(node: unknown): node is ShadowRoot {
     return !isUndefined(shadowRootRecord) && node === shadowRootRecord.shadowRoot;
 }
 
-let uid = 0;
+let uid = 1;
```

Comment:
Without this change the very first LWC element inserted into the DOM would have a `uid` of 0 and would fail [this check](https://github.com/salesforce/lwc/pull/3066/files#diff-c469e74a00289bcbdd6a87729452a408dce9319ba7516f97d319b7be64eab535R315).

Reply:
Why not not change `if (getNodeKey(context))` to `if (!isUndefined(getNodeKey(context)))` ? Then we don't need to worry about `0` vs `1`.